flixel/input/FlxSwipe.hx

```diff
@@ -3,60 +3,140 @@ package flixel.input;
 import flixel.FlxG;
 import flixel.math.FlxMath;
 import flixel.math.FlxPoint;
+import flixel.util.FlxPath;
 import flixel.util.FlxStringUtil;
 
 @:allow(flixel.input.mouse.FlxMouseButton)
 @:allow(flixel.input.touch.FlxTouch)
 class FlxSwipe
 {
+	public static var VELOCITY_THRESHOLD = 100;
+	
+	public static function filterSwipes(swipes:Array<FlxSwipe>)
+	{
+		for (swipe in swipes) 
+		{
+			if (swipe.released)
+			{
+				swipe.destroy();
+				swipes.remove(swipe);
+			}
+		}
+	}
+	
 	/**
 	 * Either LEFT_MOUSE, MIDDLE_MOUSE or RIGHT_MOUSE, 
 	 * or the touchPointID of a FlxTouch.
 	 */
 	public var ID(default, null):Int;
 	
-	public var startPosition(default, null):FlxPoint;
-	public var endPosition(default, null):FlxPoint;
+	public var path(default, null):FlxPath;
+	
+	/**
+	 * The head of `path`
+	 */
+	public var start(get, null):FlxPoint;
+	
+	/**
+	 * The tail of `path`
+	 */
+	public var end(get, null):FlxPoint;
+	
+	/**
+	 * 
+	 */
+	public var velocity(get, null):FlxPoint;
+	
+	public var released(default, null):Bool = false;
+	
+	/**
+	 * Checks if this FlxSwipe is qualifed as a swipe
+	 * Prevents mistaken tap
+	 */
+	public var qualified(get, never):Bool;
 	
+	/**
+	 * The distance from `start` to `end`
+	 */
 	public var distance(get, never):Float;
+	
 	public var angle(get, never):Float;
 	public var duration(get, never):Float;
 	
 	private var _startTimeInTicks:Float;
-	private var _endTimeInTicks:Float;
 	
-	private function new(ID:Int, StartPosition:FlxPoint, EndPosition:FlxPoint, StartTimeInTicks:Float)
+	private function new(ID:Int, StartPosition:FlxPoint, StartTimeInTicks:Float)
 	{
 		this.ID = ID;
-		startPosition = StartPosition;
-		endPosition = EndPosition;
+		path = new FlxPath();
+		path.addPoint(StartPosition);
+		velocity = FlxPoint.get();
 		_startTimeInTicks = StartTimeInTicks;
-		_endTimeInTicks = FlxG.game.ticks;
+	}
+	
+	public inline function release()
+	{
+		released = true;
+	}
+	
+	public function destroy()
+	{
+		velocity.put();
+		path.destroy();
 	}
 	
 	private inline function toString():String
 	{
-		return FlxStringUtil.getDebugString([
+		return released ? null : FlxStringUtil.getDebugString([
 			LabelValuePair.weak("ID", ID), 
-			LabelValuePair.weak("start", startPosition),
-			LabelValuePair.weak("end", endPosition),
+			LabelValuePair.weak("start", start),
+			LabelValuePair.weak("end", end),
+			LabelValuePair.weak("velocity", velocity),
 			LabelValuePair.weak("distance", distance),
 			LabelValuePair.weak("angle", angle),
-			LabelValuePair.weak("duration", duration)]);
+			LabelValuePair.weak("duration", duration),
+			LabelValuePair.weak("released", released),
+			LabelValuePair.weak("qualified", qualified)]);
+	}
+	
+	private function get_start():FlxPoint
+	{
+		return path.head();
+	}
+	
+	private function get_end():FlxPoint
+	{
+		return path.tail();
+	}
+	
+	private function get_velocity():FlxPoint
+	{
+		if (!released)
+		{
+			velocity.set(end.x - start.x, end.y - start.y);
```

Comment:
I'm not sure if this definition of velocity where you just take the start and end position into account makes sense. You could have a very fast, but _circular_ swipe, meaning that `start` and `end` are virtually the same point.